src/hooks/useTimer/index.ts

```diff
@@ -0,0 +1,83 @@
+import { useState, useRef } from 'react'
+import { TIME_IN_MS } from '../../constants'
+import { DEFAULT_INTERVAL } from './constants'
+import { StateTimer } from '../../types'
+
+const useTimer = (
+  initialTime: number,
```

Comment:
To make the hook more flexible, consider providing default values for the `initialTime`, `storeElapsedTime`, and `storeState `parameters.

```
const useTimer = (
  initialTime: number = 0,
  storeElapsedTime: number = 0,
  storeState: StateTimer | null = null,
) => { }
```